ci/common/build.sh

```diff
@@ -24,9 +24,7 @@ build_deps() {
   mkdir -p "${DEPS_BUILD_DIR}"
 
   # Use cached dependencies if $CACHE_MARKER exists.
-  if test "${CACHE_ENABLE}" = "false" ; then
-    export CCACHE_RECACHE=1
-  elif test -f "${CACHE_MARKER}" ; then
+  if test -f "${CACHE_MARKER}" && ! test "${CACHE_ENABLE}" = "false" ; then
```

Comment:
I think we can remove the second condition (`&& test "${CACHE_ENABLE}" = "false"`) since it's always set to true when using `env.sh`, and because I don't think caching can be meaningfully generalized across CI providers. If you decide to remove this then the `CACHE_ENABLE=true` in `.github/workflows/env.sh` should also be removed.